code/modules/SCP/scp.dm

```diff
@@ -1,78 +1,112 @@
-#define SAFE 1
-#define EUCLID 2
-#define KETER 3
-#define THAUMIEL 4
-#define NEUTRALIZED 5
-
 /datum/scp
-	var/name = "SCP-NULL"
-	var/designation = "0"
-	var/classification = SAFE
-	var/datum/component/scp/component //You don't have to use this, but it's really nice for smaller SCP's so you can make them SCP whenever you want
-	var/atom/owner
+	///SCP name
+	var/name
+	///SCP Designation (i.e 173 or 096)
+	var/designation
+	///SCP Class (SAFE, EUCLID, ETC.)
+	var/classification
+
+	///Meta Flags for the SCP
+	var/metaFlags
+
+	///Datum Parent
+	var/atom/parent
+
+	//Playable SCP vars
+
+	///How many players do we need on for this SCP to be playable
+	var/min_playercount = 0
+	///How much time has to have passed for this SCP to become playable
+	var/min_time = 0
+
+	//Components
+
+	///Memetic Component
+	var/datum/component/memetic/meme_comp
 
-/datum/scp/New(atom/creation)
-	creation.makeSCP()
+	//Memetic Comp Vars
+
+	///Proc called as an effect from memetic scps
+	var/memetic_proc
+	///Flags that determine how a memetic scp is detected
+	var/memeticFlags
+	///Sounds that are considered memetic
+	var/list/memetic_sounds
+
+/datum/scp/New(atom/creation, vName, vClass = SCP_SAFE, vDesg, vMetaFlags)
+	GLOB.SCP_list += creation
+
+	name = vName //names are now usually captalized improper descriptors to fit the theme of SCP since people dont just know the scp desg off the bat. As such we need to improper it. TODO: add mental mechanic for foundation workers to see desg instead of name.
+	designation = vDesg
+	classification = vClass
+	metaFlags = vMetaFlags
+
+	parent = creation
+
+	if(LAZYLEN(name))
+		parent.SetName(name)
+
+	if(classification == SCP_SAFE)
+		set_faction(parent, MOB_FACTION_NEUTRAL)
+	else
+		set_faction(parent, FACTION_SCPS)
+
+	if(ismob(parent))
+		var/mob/pMob = parent
+		if(LAZYLEN(name))
+			pMob.fully_replace_character_name(name)
+
+		if(metaFlags & SCP_PLAYABLE)
+			pMob.status_flags += NO_ANTAG
+
+	if(metaFlags & SCP_DISABLED)
+		log_and_message_staff("Disabled SCP-[designation] spawned and subsequently deleted! Do not spawn disabled SCPs!", location = get_turf(parent))
+		qdel(parent)
+		return
+
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINED, .proc/OnExamine)
+	onGain()
 
 /datum/scp/Destroy()
 	. = ..()
-	if(GLOB.SCP_list.len)
+	if(LAZYLEN(GLOB.SCP_list))
 		GLOB.SCP_list -= src
```

Comment:
The entire length check is pointless, as `-=` won't delete anything if it isn't in the list.